tests/integration/dump/test_dump_dbms.py

```diff
@@ -149,27 +154,126 @@ def error_callback(resp):
 
     dump_path = os.path.join(str(tmpdir), 'dump_dir')
     os.environ['DBMS_WORKSPACE'] = os.path.join(str(tmpdir), 'index_ws')
+    os.environ['QUERY_WORKSPACE'] = os.path.join(str(tmpdir), 'query_ws')
+
     with Flow.load_config('flow_dbms.yml') as flow_dbms:
-        with TimeContext(f'### indexing {len(docs)} docs'):
-            flow_dbms.index(docs)
+        with Flow.load_config('flow_query.yml') as flow_query:
+            with TimeContext(f'### indexing {len(docs)} docs'):
+                flow_dbms.index(docs)
+
+            with TimeContext(f'### dumping {len(docs)} docs'):
+                flow_dbms.dump('indexer_dbms', dump_path, shards=shards, timeout=-1)
 
-        with TimeContext(f'### dumping {len(docs)} docs'):
-            flow_dbms.dump('indexer_dbms', dump_path, shards=shards, timeout=-1)
+            dir_size = path_size(dump_path)
+            print(f'### dump path size: {dir_size} MBs')
 
-        dir_size = path_size(dump_path)
-        print(f'### dump path size: {dir_size} MBs')
+            # with TimeContext(f'### rolling update on {len(docs)}'):
+            #     flow_query.rolling_update('indexer_query', dump_path)
+            #
+            # flow_query.search(
+            #     docs[:nr_search],
+            #     on_done=_validate_results_nonempty,
+            #     on_error=error_callback,
+            # )
+
+            # test with query while reloading async.
+            t = Thread(
```

Comment:
I am not sure if the `multithreading` here is real. Since `flow_query.search` is blocking, the GIL may not be free and parallelism may not be real

Reply:
to be looked at

Reply:
added print statement and an assert on their order